### ISSUE TITLE
groupWith doesn't actually pass adjacent elements

### ISSUE THREAD

**Issue by henrytseng** [ramda/ramda#2262]
`groupWith` doesn't actually pass adjacent elements.  It looks like the first arg is always the first element of the last group.  

For example:
```
const list = [1,2,3,4,6,7,8,9];
const groupConsecutives = R.groupWith((a, b) => Math.abs(a - b)===1);
groupConsecutives(list); // should return [[1,2,3,4], [6,7,8,9]]
```

Returns items grouped in pairs rather than ALL consecutives grouped.  
`[ [ 1, 2 ], [ 3, 4 ], [ 6, 7 ], [ 8, 9 ] ]`

**Comment by Bradcomp** [comment]
Hi @henrytseng!

Can you let us know what version of Ramda you're using?  This was fixed for version 0.24.0. 

https://github.com/ramda/ramda/issues/2183.

Here's the test that checks for the behavior you desire: https://github.com/ramda/ramda/blob/master/test/groupWith.js#L12

**Comment by henrytseng** [comment]
Hmmmm, okay, using 0.23.0 will update
